### PR TITLE
Add support automatic changelog

### PR DESCRIPTION
**Pull request by xoofx**
Support similar to [Release Drafter](https://github.com/release-drafter/release-drafter) but built-in into `dotnet-releaser`.

The templating is using [Scriban](https://github.com/scriban/scriban)!

### CODE REVIEW

File: doc/changelog_user_guide.md [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+
+The image below is an example of an auto-generated changelog from pull-requests and commits on the project [Tomlyn](https://github.com/xoofx/Tomlyn/releases).
+
+![changelog](changelog.png)
+
+### 1.2 Properties
+
+| `[changelog]`                         | Type       | Description                |
+|---------------------------------------|------------|----------------------------|
+| `publish`                             | `bool`     | Enable to disable changelog publish. Default is `true`.
```
Comment (panmau) [inline doc/changelog_user_guide.md:43]: The description of this flag is wrong. (Enable to **disable** changelog publish)
Shouldn't it be something like: Whether changelogs should be published.

File: doc/changelog_user_guide.md [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+```md
+# Changes
+
+{{ changes }}
+
+**Full Changelog**: {{ url_full_changelog_compare_changes }}
+```
+### 1.3 Categories
+
+Categories allow to define how the changes are organized within your release notes. By default if `defaults = true` is kept `true`, the following categories will be define.
```
Comment (panmau) [inline doc/changelog_user_guide.md:72]: Probably should be something like:
- If `defaults` is kept **to** `true`, the following categories will be define**d**

Or:

- If `defaults = true` is kept, the following categories will be define**d**

File: doc/changelog_user_guide.md [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+|---------------------------------------|------------|----------------------------|
+| `title`                               | `string`   | Defines the markdown title that will be used for this category in the release notes.
+| `labels`                              | `string[]` | A list of pull-request labels that are associated with this category. Each pull-request that will have this label defined in this list will belong to this category.
+| `exclude.labels`                      | `string[]` | A list of pull-request labels that, if defined on a pull-request will exclude the pull-request to be part of this category.
+| `exclude.contributors`                | `string[]` | A list of pull-request contributors that, if are the authors in the pull-request will exclude the PR to be part of this category.
+| `order`                               | `int`      | An integer that defines the order of the category when it will be displayed in the release notes. A lower the value is, the higher in the list in the release notes the category will be displayed. If this value is not defined, it is assumed to be 0.
```
Comment (panmau) [inline doc/changelog_user_guide.md:80]: **The** lower the value is, the higher in the list in the release notes the category will be displayed

**Review by panmau — COMMENTED** [review]
I'm really excited that you made this feature happen :rocket:

Just have two small additional corrections while looking through the docs.

**Comment by xoofx** [comment]
Thank you @panmau for the feedback, this should be fixed on latest `main`

**Pull request merged**